prow/plugins/approve/approvers/owners.go

```diff
@@ -428,19 +428,21 @@ func (ap Approvers) UnapprovedFiles() sets.String {
 }
 
 // GetFiles returns owners files that still need approval.
-func (ap Approvers) GetFiles(org, repo, branch string) []File {
+func (ap Approvers) GetFiles(linkURL, org, repo, branch string) []File {
 	allOwnersFiles := []File{}
 	filesApprovers := ap.GetFilesApprovers()
 	for _, file := range ap.owners.GetOwnersSet().List() {
 		if len(filesApprovers[file]) == 0 {
 			allOwnersFiles = append(allOwnersFiles, UnapprovedFile{
+				linkURL:  linkURL,
```

Comment:
This doesn't make sense to vary between files, so I think we should refer to a more global piece of knowledge rather than add it at this level

Reply:
Agreed, how do you guys feel about me packing these items in `github.Client`? Something like this:
```
// Client interacts with the github api.
type Client struct {
    // If logger is non-nil, log all method calls with it.
    logger *logrus.Entry
    time   timeClient

    gqlc       gqlClient
    client     httpClient
    bases      []string
    linkURL    string
    apiLinkURL string
    dry        bool
    fake       bool
    throttle   throttler
    getToken func() []byte

    mut     sync.Mutex // protects botName and email
    botName string
    email   string
}
```

I think that makes more sense, as then the plugins rely on the github client for these settings.

I also want to move the `GetFiles` url building logic into github client, does that sound like a good idea?

Reply:
Please address this.

Reply:
This was the easiest injection point I could find, by just injecting the variable where the static `https://github.com` value was used. These files are iterated over to produce the comment template here: https://github.com/kubernetes/test-infra/blob/3f690e0b16bde5f935e454f6e5c1f6e21c944a65/prow/plugins/approve/approvers/owners.go#L648

I can't think of a way to override this value without changing template creation logic. Is that something we would like to do now, or can limit change and introduce the override as close to the existing static value implementation as possible?

Thoughts?

Reply:
Why can't we just pass it from:

 - `handle()` (by accessing config)
 - `updateNotification()`
 - `approvers.GetMessage()`

It's static data like the org and repo.

Reply:
org and repo are also passed at the file level, like this:
https://github.com/kubernetes/test-infra/blob/60c608651870d71fe7aa13e3a6e4e909aa079e0d/prow/plugins/approve/approvers/owners.go#L439

And they are part of UnapprovedFile and ApprovedFile: https://github.com/kubernetes/test-infra/blob/60c608651870d71fe7aa13e3a6e4e909aa079e0d/prow/plugins/approve/approvers/owners.go#L558

This was probably done this way to allow template changes in the stringer functions: https://github.com/kubernetes/test-infra/blob/60c608651870d71fe7aa13e3a6e4e909aa079e0d/prow/plugins/approve/approvers/owners.go#L575

Do we want to re-work these template functions, or can we leave that for another PR?

Reply:
Let's keep that change in a follow-up!